### PR TITLE
Tickless fixes

### PR DESCRIPTION
**Pull request by vext01**
The first commit message should help to explain the need for this change.

The second commit is just test housekeeping. Also note that the test suite explodes badly using newer py.test versions. I had to install the old versions listed in `requirements.txt`.

### CODE REVIEW

**Comment by ltratt** [comment]
I am not worried about Python 2. It will be around for a fair while yet and porting Krun would take more effort than I think we can justify at the moment.

File: krun/platform.py [hunk 1]
```diff
@@ -1036,9 +1029,22 @@ def _check_tickless_kernel(self):
 
         debug("Checking linux kernel is tickless")
 
+        if self.num_cpus == 1:
+            # Tickless mode doesn't make sense on a single-cpu system, as you
+            # need at least on non-adaptive ticks CPU to do various
```
Comment (ltratt) [inline krun/platform.py]: "at least on non-adaptive" doesn't parse. [I can guess the overall meaning, but only because I know the area!]

Comment (vext01) [reply krun/platform.py]: ah, `on` -> `one`. Good catch.

Comment (ltratt) [reply krun/platform.py]: Even then it's still a bit hard to parse. Maybe "non-adaptive-ticks" helps?

**Comment by vext01** [comment]
> I am not worried about Python 2. It will be around for a fair while yet and porting Krun would take more effort than I think we can justify at the moment.

I think you are right. I just checked Debian testing and that is surprisingly still Python 2 by default. So that means Debian 11 (when it comes) will still use Python 2 as default.

**Comment by vext01** [comment]
Pushed a fix.

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat. also updated PR description.
